### ISSUE TITLE
Oxy 4.7.1 PHP Deprecated  'strpos()/str_replace(): Passing null to parameter ...' in Backend

### ISSUE THREAD

**Issue by maltmann-muc** [soflyy/oxygen-bugs-and-features#3452]
**Environment:** PHP 8.1, WP 6.4.1, Oxy 4.7.1 
**Situation:** In general for every backend screen

**Files and Functions:**
wp-content/plugins/oxygen/component-framework/admin/pages.php
• ct_admin_settings()
• oxygen_vsb_register_signing_page()
wp-content/plugins/oxygen/component-framework/includes/oxygen-connection.php
• oxygen_vsb_connection_register_options_page()

**Warning and Call Stack:**
(Sample for one of the three cases)
```
Deprecated:  strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated in /.../wp-includes/functions.php on line 7241
Stack trace:
  1. {main}() /.../wp-admin/index.php:0
  2. require_once() /.../wp-admin/index.php:10
  3. require() /.../wp-admin/admin.php:158
  4. require_once() /.../wp-admin/menu.php:417
  5. do_action($hook_name = 'admin_menu', ...$arg = variadic('')) /.../wp-admin/includes/menu.php:161
  6. WP_Hook->do_action($args = [0 => '']) /.../wp-includes/plugin.php:517
  7. WP_Hook->apply_filters($value = '', $args = [0 => '']) /.../wp-includes/class-wp-hook.php:348
  8. oxygen_vsb_register_signing_page('') /.../wp-includes/class-wp-hook.php:324
  9. add_submenu_page($parent_slug = NULL, $page_title = 'Oxygen Sign Shortcodes', $menu_title = 'Oxygen Sign Shortcodes', $capability = 'read', $menu_slug = 'oxygen_vsb_sign_shortcodes', $callback = 'oxygen_vsb_sign_shortcodes_page', $position = *uninitialized*) /.../wp-content/plugins/oxygen/component-framework/admin/pages.php:1674
 10. plugin_basename($file = NULL) /.../wp-admin/includes/plugin.php:1409
 11. wp_normalize_path($path = NULL) /.../wp-includes/plugin.php:769
 12. wp_is_stream($path = NULL) /.../wp-includes/functions.php:2180
 13. strpos($haystack = NULL, $needle = '://') /.../wp-includes/functions.php:7241

Deprecated:  str_replace(): Passing null to parameter #3 ($subject) of type array|string is deprecated in /.../wp-includes/functions.php on line 2187
Stack trace:
  1. {main}() /.../wp-admin/index.php:0
  2. require_once() /.../wp-admin/index.php:10
  3. require() /.../wp-admin/admin.php:158
  4. require_once() /.../wp-admin/menu.php:417
  5. do_action($hook_name = 'admin_menu', ...$arg = variadic('')) /.../wp-admin/includes/menu.php:161
  6. WP_Hook->do_action($args = [0 => '']) /.../wp-includes/plugin.php:517
  7. WP_Hook->apply_filters($value = '', $args = [0 => '']) /.../wp-includes/class-wp-hook.php:348
  8. oxygen_vsb_register_signing_page('') /.../wp-includes/class-wp-hook.php:324
  9. add_submenu_page($parent_slug = NULL, $page_title = 'Oxygen Sign Shortcodes', $menu_title = 'Oxygen Sign Shortcodes', $capability = 'read', $menu_slug = 'oxygen_vsb_sign_shortcodes', $callback = 'oxygen_vsb_sign_shortcodes_page', $position = *uninitialized*) /.../wp-content/plugins/oxygen/component-framework/admin/pages.php:1674
 10. plugin_basename($file = NULL) /.../wp-admin/includes/plugin.php:1409
 11. wp_normalize_path($path = NULL) /.../wp-includes/plugin.php:769
 12. str_replace($search = '\\', $replace = '/', $subject = NULL) /.../wp-includes/functions.php:2187
```

**Description:**
Oxygen calls add_submenu_page() to add "invisible" menu items 'Add 3rd Party Design Set', 'Oxygen Sign Shortcodes', 'Oxygen QA Check', passing null as first parameter.
 First parameter to add_submenu_page is a required string.
<img width="958" alt="Bildschirmfoto 2023-11-19 um 10 49 20" src="https://github.com/soflyy/oxygen-bugs-and-features/assets/51136650/cd95c9a8-8113-4756-bd7d-6a9b94c61908">

Solution:
For all three occurrences, change
add_submenu_page(null, ...
to
add_submenu_page('null', ...

**Comment by maltmann-muc** [comment]
Oxy 4.8 RC 1: Issue persists, but reporting for menu item "Oxygen QA Check" seems gone.

**Comment by maltmann-muc** [comment]
Oxy 4.8: Issue persists,

**Comment by Kpudlo** [comment]
Confirmed and reported internally via https://github.com/soflyy/oxygen/issues/5187

**Comment by maltmann-muc** [comment]
Oxy 4.8.3: Issue persists.

**Comment by FORTE-WP** [comment]
Apparently can be solved easily by replacing the parent slug (null) in add_submenu_page by an empty string (''). That's at least what Yoast has done, and which was validated by jrfnl: https://github.com/Yoast/wordpress-seo/pull/19125. Seems to be working fine on my development server.

**Comment by maltmann-muc** [comment]
> Apparently, this can be solved easily by replacing the parent slug (null) in add_submenu_page with an empty string (''). 

Yes, absolutely. The first parameter must be a string, not `null`.

**Comment by FORTE-WP** [comment]
Sorry @maltmann-muc, just seconds before your response I removed my comment, because my solution to replace the parent slug null by an empty string as Yoast has done, created a new error. The empty string returned: strip_tags(): Passing null to parameter #1 ($string) of type string is deprecated.

**Comment by maltmann-muc** [comment]
> Sorry @maltmann-muc, just seconds before your response I removed my comment, because my solution to replace the parent slug null by an empty string as Yoast has done, created a new error. The empty string returned: strip_tags(): Passing null to parameter #1 ($string) of type string is deprecated (wp-admin/admin-header.php:36)

I can't imagine why an empty string should cause any issues. An empty string is not `null`. Maybe I find some time to investigate


In my request I suggested `'null'` (as string) which doesn't cause any issues – since at least a year where I used this modification.

**Comment by FORTE-WP** [comment]
The error occurs when accessing the submenu pages that have an empty parent slug. No title, not even an empty string, is returned by the function get_admin_page_title() (wp-admin/admin-header.php:35). Using any fake slug will probably do the trick. BTW: Yoast has the same problem for submenu pages with empty strings parent slugs.

**Comment by maltmann-muc** [comment]
> The error occurs when accessing the submenu pages that have an empty parent slug. ... Using any fake slug will probably do the trick. 

Thank you for reporting.
Then my initial suggestion using `'null'` (= fake slug) will work – as I can confirm from experiences.

**Comment by perrelet** [comment]
@Spellhammer  Why does this remain unfixed?

**Comment by maltmann-muc** [comment]
Oxy 4.9: Issue persists.

**Comment by perrelet** [comment]
How do you guys get away with this?

![image](https://github.com/user-attachments/assets/deb69f15-9367-47cc-87c2-2eacb276df9e)

**Comment by maltmann-muc** [comment]
Oxy 4.9.1: Issue still persists. After over a year...

**Comment by perrelet** [comment]
Once again, @soflyy in the habit of deleting comments it doesn't like. 

![image](https://github.com/user-attachments/assets/8a1d7009-47f3-44a0-a97c-ece3cd036845)

**Comment by perrelet** [comment]
Can someone at @soflyy please comment on this issue. Thanks.

**Comment by gik000** [comment]
I also am experiencing this issue since long time :|
Can you please provide some feedback?

**Comment by perrelet** [comment]
Since it seems highly unlikely we'll be hearing from soflyy anytime soon, on old sites still running oxy, I install [this mu-plugin](https://gist.github.com/perrelet/93bffcd8e8f3e5b69e0c2fd1b49a5041) to suppress all deprecation notices. A bit of hammer nevertheless, but it does silence this garbage 🗑️🔨

**Comment by maltmann-muc** [comment]
Finally fixed in 4.9.4

**Comment by perrelet** [comment]
lol
